src/Illuminate/Console/Scheduling/Schedule.php

```diff
@@ -25,6 +39,10 @@ class Schedule
     public function __construct(Cache $cache)
     {
         $this->cache = $cache;
+
+        $this->phpBinary = ProcessUtils::escapeArgument((new PhpExecutableFinder)->find(false));
+
+        $this->artisan = defined('ARTISAN_BINARY') ? ProcessUtils::escapeArgument(ARTISAN_BINARY) : 'artisan';
```

Comment:
I think it would be better to extract this to some static method on a class or something that way we don't have to do all this just to pass a command name into the event class.